### ISSUE TITLE
Email notifications: signer modal component and transaction

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/pioneer#4232]
## Context
- https://github.com/Joystream/pioneer/issues/4230

## Backend
⚠️ [Link to BE API implementation >](https://github.com/Joystream/pioneer/blob/backend/packages/server/README.md)

## Dependencies

Blocks: 
- #4231 
- #4237 

Blocked by:
- #4274 

## Scope
- [ ] Create component (Signer modal for connecting tx email with membership.) [Link to Design > ](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-design?node-id=11947%3A511977&t=kbnLvo8CmtA4rR1I-1)
-- email input
-- membership controller account displayed which is used to sign
-- Ctas: "sign and authorize email"; "Not now"
-- state when email is not provided -> CTA  "sign and authorise" is disabled unless email is provided
- [ ] When email is provided and modal signed, confirmation email must be sent.. 
- [ ] Confirmation email template for email auth. HTML template is required to be used  see
- #4274 point 1.
- [ ] Following the link user is brought back to the App with confirmation modal shown. [Link to design> ](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-design?node-id=11947%3A508551&t=kbnLvo8CmtA4rR1I-1)

## Design
Links are added to acceptance criteria
Preview: 
<img width="633" alt="Screenshot 2023-03-30 at 18 08 42" src="https://user-images.githubusercontent.com/93924525/228863392-60e3454a-f765-4f31-88db-fce5c5be502b.png">

**Comment by IlyaSmiyukha** [comment]
I have a few questions about styling the modal. 

1. I didn't find text for the tooltip in label
2. Can I remove the such a big gap between footer and blue alert? 
3. I see `background-color: #E8EDF6;` for footer in figma, but default color is #fff for all modals. Should I change it?

**Comment by dmtrjsg** [comment]
@IlyaSmiyukha 
Thanks for the questions.

1. Tooltip text: Add your email address here to receive notifications. It can be the same or different to the one added to the membership profile.
2. Let's keep the modal dimensions as this component is standardized. 
3. Feel free to use the default colour for the modal footers.

**Comment by IlyaSmiyukha** [comment]
@dmtrjsg Got it. Thanks for the quick response

**Comment by dmtrjsg** [comment]
@IlyaSmiyukha have you managed to run the backend to be able to test this?

**Comment by IlyaSmiyukha** [comment]
@dmtrjsg No, I got this error. But .env I copied from .env.dev
<img width="1098" alt="Screenshot 2023-04-08 at 14 31 46" src="https://user-images.githubusercontent.com/39830587/230718852-4cf1fbc3-22a7-4212-9064-5b729640c23e.png">

**Comment by dmtrjsg** [comment]
@IlyaSmiyukha ok thx for letting us know i'll try to resolve asap. Monday is a day off for most of our team so perhaps this would need to wait till Tuesday morning.

**Comment by thesan** [comment]
> @dmtrjsg No, I got this error. But .env I copied from .env.dev
> <img width="1098" alt="Screenshot 2023-04-08 at 14 31 46" src="https://user-images.githubusercontent.com/39830587/230718852-4cf1fbc3-22a7-4212-9064-5b729640c23e.png">
> 

@IlyaSmiyukha this means that you shouldn't define both the sendgrid **and** the mailgun config. But for the purpose of this issue, you don't need to define an email provider. So all the variables in the _Email (only one of the api provider should be defined)_ section should be removed or commented from your `.env` file.

**Comment by IlyaSmiyukha** [comment]
@thesan it helped, thanks

**Comment by thesan** [comment]
@IlyaSmiyukha np !

I just fixed issues with the sign-up/sign-in mutations on the branch of this PR: #4337. So you should run the backend from this branch to implement this modal.
